lumicks/pylake/kymotracker/detail/msd_estimation.py

```diff
@@ -1000,22 +1023,26 @@ def ensemble_cve(kymotracks):
     """
     cve_based = kymotracks.estimate_diffusion(method="cve")
 
+    def mean_var(x, count, count_sum):
+        # Equation 58 and 57 from Vestergaard et al [14].
+        weighted_mean = np.sum(x * count) / count_sum
+        weighted_mean_var = np.sum(count * (x - weighted_mean) ** 2) / (
+            (count.size - 1) * count_sum
+        )
+        return weighted_mean, weighted_mean_var
+
     if len(cve_based) == 1:
         return cve_based[0]
 
     estimates = np.array([c.value for c in cve_based])
     counts = np.array([c.num_points for c in cve_based])
-    counts_sum = np.sum(counts)
+    counts_sum = sum(counts)
 
-    # Equation 58 and 57 from Vestergaard et al [14].
-    ensemble_mean = np.sum(estimates * counts) / counts_sum
-    ensemble_mean_var = np.sum(counts * (estimates - ensemble_mean) ** 2) / (
-        (counts.size - 1) * counts_sum
+    ensemble_mean, ensemble_mean_var = mean_var(estimates, counts, counts_sum)
+    ensemble_localization_variance, ensemble_localization_var_var = mean_var(
+        np.array([c.localization_variance for c in cve_based]), counts, counts_sum
```

Comment:
Nitpick for consistency: Either define `localization_variance = np.array([c.localization_variance for c in cve_based])` above and use it here or avoid defining `estimates = np.array([c.value for c in cve_based])`

Reply:
I'll inline it. Seems more consistent indeed.

Reply:
👍